### DISCUSSION TITLE
Component testing with Astro

### DISCUSSION THREAD

**Discussion by nicolalandro** [cypress-io/cypress#25746] [Questions and Help]
I reach to testing end2end [astro](https://astro.build/) app, but when I try to configure componente test with svelte I have a problem on the budler because I cannot specify astro, also if in truth astro use vite. By specifiing vite he do not find the vite config file.

How can I test my astro or svelte component under Astro with cypres? Can I specify custom bundler?

**Comment by portyu9** [comment]
Hi @nicolalandro 

For a `.svelte` component inside an Astro project, configure Cypress as **Svelte + Vite**, not as Astro:

```js
import { defineConfig } from 'cypress'

export default defineConfig({
  component: {
    devServer: {
      framework: 'svelte',
      bundler: 'vite',
    },
  },
})
```

Current Cypress can use an explicit `viteConfig` if there is no standalone `vite.config.*` for it to discover:

```js
export default defineConfig({
  component: {
    devServer: {
      framework: 'svelte',
      bundler: 'vite',
      viteConfig: {
        // aliases/plugins/options needed by your components
      },
    },
  },
})
```

This distinction matters because meta-frameworks can configure Vite internally without exposing a normal `vite.config` file for Cypress to discover.

For actual **`.astro` components**, Cypress still does not have a first-party Astro mounting integration. The current built-in framework values are React, Vue, Svelte, Next.js and Angular.

So:

- test `.svelte` components as Svelte components with the Vite bundler;
- test full Astro pages through Cypress E2E;
- for direct `.astro` component mounting, you would need a community/custom Cypress Component Testing framework adapter.

You can supply a community/custom **framework definition**, but the built-in bundler choices remain Vite and Webpack.

Please mark this answer as accepted if it helped, thank you.
